### ISSUE TITLE
Colored `ProgressBar`

### ISSUE THREAD

**Issue by HenryTheCat** [gyscos/Cursive#278]
This is a feature request proposing an addition to the `ProgressBar` view.

# Problem description

Currently, `ProgressBar` uses the theme's `highlight` color as the bar's color, and that is not customizable. It may be desiderable to have progress bars with customized bar colors.

# Proposal

Extend the `ProgressBar` API in a backward-compatible way to optionally specify a custom color.

As a tentative implementation, a `custom_color: Option<Color>` field could be added to `ProgressBar`, initialized as `None` by default. New methods could be added to modify the new field. Then, when printing, `ProgressBar` would use its custom color if specified, or the theme's `highlight` otherwise.

**Comment by HenryTheCat** [comment]
The issue looks simple enough that I may try implementing it myself, if the proposal is approved.

**Comment by gyscos** [comment]
Hi, and thanks for the idea!

Adding such a custom color could be an option (though you may want to take a `ColorType` to let people use palette colors if they need).

Another option is something we've been working toward recently, which is locally overriding the theme. The idea is to derive a new theme just for this `ProgressBar`, by taking the parent theme and replacing the `Highlight` color from the palette with a new custom color. This effectively achieves the same result, but is a bit more general as it lets users override any color for any view.

The main issue with this solution is that users need to know _what_ color to override: they would need to read the documentation, see the mention of `Highlight`, and override this color. This is all a bit less obvious than an inherent method.

Note that these two methods are not incompatible - we may add a `custom_color` to `ProgressBar`, and still let users apply local themes overrides. This may be a way to see how convenient it is in real conditions.

TLDR: let's do it!
